### ISSUE TITLE
Support connecting to hard-coded server

### ISSUE THREAD

**Issue by Benau** [supertuxkart/stk-code#2446]
Me and @Odd0002 tested with this "hack":
https://gist.github.com/Benau/ad9d501273a37e9b7c0d (not needed now, please use the branch)

Actually it only crashed the first time run, the next run worked.
Which means, Wan game using lan method without STUN works!

It would be good to allow client to connect to STK server with public ip.(ie: without STUN protocol and STK addon server)

**Comment by Benau** [comment]
For the hack in PingProtocol:

I think there should be no need to call this in connect to server, as m_server_address will always know (unless hidden which is handled in HIDING_ADDRESS) before getting in GOT_SERVER_ADDRESS.

so:
```
        case REQUESTING_CONNECTION:
        {
            // In case of a LAN server, m_current_protocol is NULL
            if (!m_current_protocol ||
                m_current_protocol->getState() == PROTOCOL_STATE_TERMINATED)
            {
                delete m_current_protocol;
                m_current_protocol = NULL;
                // Server knows we want to connect
                Log::info("ConnectToServer", "Connection request made");
                if (m_server_address.getIP() == 0 ||
                    m_server_address.getPort() == 0  )
                {
                    // server data not correct, hide address and stop
                    m_state = HIDING_ADDRESS;
                    Log::error("ConnectToServer", "Server address is %s",
                               m_server_address.toString().c_str());
                    m_current_protocol = new HidePublicAddress();
                    m_current_protocol->requestStart();
                    return;
                }
                handleSameLAN();
            }
            break;
        }
```
should work

Also pingprotocol is used in connect_to_peer too, it says: 
```
            // Now we know the peer address. If it's a non-local host, start
            // the Ping protocol to keep the port available. We can't rely on
            // STKHost::isLAN(), since we might get a LAN connection even if
            // the server itself accepts connections from anywhere.
            if (!m_is_lan &&
                m_peer_address.getIP() != NetworkConfig::get()
                                          ->getMyAddress().getIP())
            {
                m_current_protocol = new PingProtocol(m_peer_address,
                                                      /*time-between-ping*/2.0);
                ProtocolManager::getInstance()->requestStart(m_current_protocol);
                m_state = CONNECTING;
                break;
            }
```
I think mainly it's used to keep the port open to peer, as it could be behind a router, but the lan server (even the server I hardcoded for tested) should not behind a router anway, so I think it should not be used in connect_to_server, unless it's wan server with stun (But I'm not sure how to add pingprotocol for wan server stun there).

Notice: The test with @Odd0002: he's using wifi to connect to my public server, so seems that only one server is needed to be public, we might not need to bother the client type.

**Comment by Benau** [comment]
I created a new branch called fix_2446, now you can connect to a lan(if it's public ip it will work for wan too) server in command line specified, using supertuxkart --connect-now=1.2.3.4:1234 format.

tested using 2 PCs with real ip in linux(s).

**Comment by qwertychouskie** [comment]
Maybe this is a dumb question (sorry if it is), but if it works, why not merge it?

**Comment by auriamg** [comment]
@qwertychouskie I know you're well intended but could you please focus your activity on only the tickets where you actually have something important to add? Just asking questions or making little comments is not helping, I get a mail in my inbox for every comment you post and maybe half of them bring nothing of actual interest to the actual issue. Thanks

**Comment by hiker** [comment]
This was merged in b0294060dcbd11d2fb3a671316fdd6d81d44f0aa
